test/unit/sdam/topology.test.js

```diff
@@ -314,5 +319,108 @@ describe('Topology (unit)', function () {
       });
       return p;
     });
+
+    describe('srv event listeners', function () {
+      /** @type {Topology} */
+      let topology;
+
+      beforeEach(() => {
+        topology = new Topology('', { srvHost: 'fakeHost' });
+
+        expect(topology.s.detectSrvRecords).to.be.a('function');
+        expect(topology.s.detectShardedTopology).to.be.a('function');
+      });
+
+      function transitionTopology(topology, from, to) {
+        topology.emit(
+          Topology.TOPOLOGY_DESCRIPTION_CHANGED,
+          new TopologyDescriptionChangedEvent(
+            2,
+            new TopologyDescription(from),
+            new TopologyDescription(to)
+          )
+        );
+        // We don't want the SrvPoller to actually run
+        clearTimeout(topology.s.srvPoller._timeout);
+      }
+
+      describe('srvRecordDiscovery event listener', function () {
+        beforeEach(() => {
+          // fake a transition to Sharded
+          transitionTopology(topology, TopologyType.Unknown, TopologyType.Sharded);
+          expect(topology.s.srvPoller).to.be.instanceOf(SrvPoller);
+
+          const srvPollerListeners = topology.s.srvPoller.listeners(SrvPoller.SRV_RECORD_DISCOVERY);
+          expect(srvPollerListeners).to.have.lengthOf(1);
+          expect(srvPollerListeners[0]).to.equal(topology.s.detectSrvRecords);
+          const topologyChangeListeners = topology.listeners(Topology.TOPOLOGY_DESCRIPTION_CHANGED);
+          expect(topologyChangeListeners).to.have.lengthOf(1);
+          expect(topologyChangeListeners[0]).to.equal(topology.s.detectShardedTopology);
+        });
+
+        it('should emit topologyDescriptionChange event', function () {
+          topology.once(Topology.TOPOLOGY_DESCRIPTION_CHANGED, ev => {
+            // The first event we get here is caused by the srv record discovery event below
+            expect(ev).to.have.nested.property('newDescription.servers');
+            expect(ev.newDescription.servers.get('fake:2'))
+              .to.be.a('object')
+              .with.property('address', 'fake:2');
+          });
+
+          topology.s.srvPoller.emit(
+            SrvPoller.SRV_RECORD_DISCOVERY,
+            new SrvPollingEvent([{ priority: 1, weight: 1, port: 2, name: 'fake' }])
+          );
+
+          // The srv event starts a monitor that we don't actually want running
```

Comment:
let's move the cleanup into an afterEach and iterate over all servers

Reply:
Yes, good idea 🚀